### DISCUSSION TITLE
Combining two files with fields overriding entirely on the other file.

### DISCUSSION THREAD

**Discussion by akshatdalton** [mikefarah/yq#1464] [Q&A]
Hello there, I'm new here. Thank you for making this amazing software.
I have a question that I want to resolve. (I hope this is the best place to ask, if not then please point it out to me.)
Suppose I have a file:
`file1.yaml` with contents:
```.yaml
a:
  b:
    c:
      e: 1
      f: 2
    d: foo
  g: bar
h: baz
```

and `file2.yaml` with contents:
```.yaml
a:
  b:
    c:
      g: 3
```

when I run: `yq eval-all '. as $item ireduce ({}; . * $item )' file2.yaml file1.yaml`, it outputs:
```.yaml
a:
  b:
    c:
      e: 1
      f: 2
      g: 3
    d: foo
  g: bar
h: baz
```

But I want the fields present under `c` to complete change from the contents of the `file2.yaml`, i.e. I want the output to be:
```.yaml
a:
  b:
    c:
      g: 3
    d: foo
  g: bar
h: baz
```

So, I want the values present under one field to completely change if the contents under that field is changed in another file 2.

I can achieve the same by doing: `yq '.a.b.c = {"g": 3}' file1.yaml` but I'm dealing with two files at a time and their contents are dynamic; their merge action should result in output something like above.

**Comment by mikefarah** [comment]
Yeah there's not way to do that (as far as I know) without explicitly doing what you had '.a.b.c = {"g": 3}'.
